packages/openapi3/src/openapi.ts

```diff
@@ -817,6 +818,24 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
       description: getDoc(program, model),
     };
 
+    const discriminator = getDiscriminator(program, model);
```

Comment:
see my review on the cadl-azure pr.  I feel like this functionality ought to be moved into the extension defining the decorator, so it isn't duplicated across emitters.  I think both getting the discriminators, getting the child types, and getting the value -> type mapping shpould be common functions.    However, I don't think we need to hold up the PR for this.